src/ui/public/query_bar/components/query_bar.tsx

```diff
@@ -137,6 +143,10 @@ export class QueryBar extends Component<Props, State> {
   private componentIsUnmounting = false;
   private persistedLog: PersistedLog | null = null;
 
+  public isDirty = () => {
+    return this.state.query.query !== this.props.query.query;
```

Comment:
Should we do a check for the `language` here too? What if you just changed it in the url?

Reply:
JK, I guess if that happens then a refresh will happen anwyay